### PR TITLE
TC_15_PlaceOrder_RegisterBeforeCheckout

### PR DESCRIPTION
**Pull request by basant-lazem**

### CODE REVIEW

**Comment by basant-lazem** [comment]
Pipeline : 
https://github.com/MahmoudElSharkawyGS/AutomationExercisePractice/actions/runs/8002376959

**Comment by basant-lazem** [comment]
Pipeline : 
https://github.com/MahmoudElSharkawyGS/AutomationExercisePractice/actions/runs/8003036989

File: src/main/java/com/gizasystems/automationexercise/pages/CartPage.java [hunk 1]
```diff
@@ -7,11 +7,10 @@
 public class CartPage {
     // Variables
     private SHAFT.GUI.WebDriver driver;
-
     // Locators
     private final By proceedToCheckout_btn = By.cssSelector(".btn.btn-default.check_out");
     private final By productName_h4 = By.xpath("//td[@class='cart_description']//h4");
-    private final By viewCart_a = By.xpath("//div[contains(@class,'confirm')]//a[@href='/view_cart']");
+    private final By viewCart_a = By.cssSelector("#header > div > div > div > div.col-sm-8 > div > ul > li:nth-child(3) > a");
```
Comment (MahmoudElSharkawyGS) [inline src/main/java/com/gizasystems/automationexercise/pages/CartPage.java:13]: This is not a relative path, why did you change it ? :D

File: src/main/java/com/gizasystems/automationexercise/pages/CheckOutPage.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package com.gizasystems.automationexercise.pages;
+import com.shaft.driver.SHAFT;
+import io.qameta.allure.Step;
+import org.openqa.selenium.By;
+public class CheckOutPage {
+    private SHAFT.GUI.WebDriver driver;
+    // Locators
+    private String url = System.getProperty("baseUrl") + "/checkout";
+    private final By addressVerifiycation_FirstName = By.xpath("//ul[@class = \"address item box\"]//li[@class=\"address_firstname address_lastname\"]");
+    private  final By addressVerifiycation_address1 = By.xpath("/html/body/section/div/div[3]/div/div[1]/ul/li[4]");
```
Comment (MahmoudElSharkawyGS) [inline src/main/java/com/gizasystems/automationexercise/pages/CheckOutPage.java:10]: This is not a relative path, please enhance it to be more relative.

File: src/main/java/com/gizasystems/automationexercise/pages/CheckOutPage.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    @Step ("Verifiying address details")
+    public CheckOutPage verififyingAddressDetails(String FirstName , String Address , String City){
+        driver.verifyThat().element(addressVerifiycation_FirstName).text().equalsIgnoringCaseSensitivity(FirstName).perform();
+        driver.verifyThat().element(addressVerifiycation_address1).text().equalsIgnoringCaseSensitivity(Address).perform();
+        driver.verifyThat().element(addressVerifiycation_city).text().equalsIgnoringCaseSensitivity(City).perform();
+        return  this;
+    }
+
```
Comment (MahmoudElSharkawyGS) [inline src/main/java/com/gizasystems/automationexercise/pages/CheckOutPage.java:29]: Validation steps should be at the bottom on the class as agreed.

File: src/main/java/com/gizasystems/automationexercise/pages/PaymentPage.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        return this;
+    }
+    @Step("Click 'Pay and Confirm Order' button")
+    public PaymentPage clickOnPayOrder() {
+        driver.element().click(payOrder_Btn);
+        return this;
+    }
+
+    @Step("Verify success message 'Your order has been placed successfully!'")
+    public PaymentPage VerifySucessMessage(String Message){
```
Comment (MahmoudElSharkawyGS) [inline src/main/java/com/gizasystems/automationexercise/pages/PaymentPage.java:42]: Validation steps should be at the bottom on the class as agreed.

File: src/main/java/com/gizasystems/automationexercise/pages/SignupPage.java [hunk 5]
```diff
@@ -73,8 +72,7 @@ public SignupPage enterAddressInformation(String userAddressOne, String country,
 
     @Step("Click on Continue Button")
     public SignupPage clickOnContinueButton() {
-        driver.element().click(continueBtn_link);
-        GoogleAlert.dismissAlert(driver, continueBtn_link);
+        driver.element().clickUsingJavascript(continueBtn_link);
         return this;
```
Comment (MahmoudElSharkawyGS) [inline src/main/java/com/gizasystems/automationexercise/pages/SignupPage.java:76]: Why did you change this logic?

File: src/test/java/com/gizasystems/automationexercise/tests/test/PlaceOrder_RegisterBeforeCheckout.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+public class PlaceOrder_RegisterBeforeCheckout {
+    private SHAFT.TestData.JSON testData;
+    private SHAFT.TestData.JSON productData;
+    private SHAFT.TestData.JSON CardData;
+    private SHAFT.GUI.WebDriver driver;
+    @BeforeClass
+    public void beforeClass() {
+        testData = new SHAFT.TestData.JSON("src/test/resources/testDataFiles/RegisterUser.json");
+        productData = new SHAFT.TestData.JSON("src/test/resources/testDataFiles/Product.json");
+        CardData = new SHAFT.TestData.JSON("src/test/resources/testDataFiles/CardData.json");
```
Comment (MahmoudElSharkawyGS) [inline src/test/java/com/gizasystems/automationexercise/tests/test/PlaceOrder_RegisterBeforeCheckout.java:26]: We only need one test data json path as agreed, why do you have 3?

File: src/test/java/com/gizasystems/automationexercise/tests/test/PlaceOrder_RegisterBeforeCheckout.java [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+                .verififyingAddressDetails("Mr. Automation Bot",testData.getTestData("UserAddress1"),testData.getTestData("UserCountry"))
+                .enteringDescriptionInCommentArea( "Place Order");
+        new PaymentPage(driver)
+                .navigate()
+                .enterPaymentDetails(CardData.getTestData("CardName"),CardData.getTestData("CardNumber") , CardData.getTestData("CVC"), CardData.getTestData("CardExpMonth"),CardData.getTestData("CardExpYear"))
+                .clickOnPayOrder()
+                .VerifySucessMessage(CardData.getTestData("SuccessMessage"))
+                .DelteAccount();
+        new DeleteAccountPage(driver)
+                .validateAccountDeleted("ACCOUNT DELETED!");
```
Comment (MahmoudElSharkawyGS) [inline src/test/java/com/gizasystems/automationexercise/tests/test/PlaceOrder_RegisterBeforeCheckout.java:72]: test data should not be static in the test like this. Please move this to the test data file.

**Review by MahmoudElSharkawyGS — CHANGES_REQUESTED** [review]
- Please follow the naming convention (lowerCamaelCase) for variables
- Please enhance the locators :D 
- Please make sure to reformat your code.

File: src/main/java/com/gizasystems/automationexercise/pages/CartPage.java (hunk 1 above)
Comment (basant-lazem) [reply src/main/java/com/gizasystems/automationexercise/pages/CartPage.java:13]: It only passed with this cssSelector

File: src/test/java/com/gizasystems/automationexercise/tests/test/PlaceOrder_RegisterBeforeCheckout.java (hunk 6 above)
Comment (basant-lazem) [reply src/test/java/com/gizasystems/automationexercise/tests/test/PlaceOrder_RegisterBeforeCheckout.java:26]: To make all of them accessible to my TC as  I test a full Scenario

File: src/main/java/com/gizasystems/automationexercise/pages/CartPage.java (hunk 1 above)
Comment (MahmoudElSharkawyGS) [reply src/main/java/com/gizasystems/automationexercise/pages/CartPage.java:13]: I don't think so :D. Please share the error and let us know to handle it better. We will not be using absolute paths :D

File: src/test/java/com/gizasystems/automationexercise/tests/test/PlaceOrder_RegisterBeforeCheckout.java (hunk 6 above)
Comment (MahmoudElSharkawyGS) [reply src/test/java/com/gizasystems/automationexercise/tests/test/PlaceOrder_RegisterBeforeCheckout.java:26]: We agreed to have 1 to 1 relationship between a test class and a test data json file to make the code and test classes more independent on each other's and enhance the readability and scalability. Please make sure to apply the conventions we agreed on.

**Review by MahmoudElSharkawyGS — CHANGES_REQUESTED** [review]
- Please make sure to reformat your code.
- Also, please make sure to execute the pipeline again and share the link here.

**Review by MahmoudElSharkawyGS — CHANGES_REQUESTED** [review]

**Review by MahmoudElSharkawyGS — APPROVED** [review]
